R/db.R

```diff
@@ -159,8 +159,8 @@ write_data_safely <- function(dbconn, tblname, df, key) {
     df[, id_cols] <- df[, id_cols_ix]
 
     # Convert some types to character so they go in the DB properly.
-    to_chars <- vapply(df, function(x) is.factor(x) || is.ordered(x) || is.logical(x), logical(1))
-    df[, to_chars] <- lapply(df[, to_chars], as.character)
+    to_chars <- unname(vapply(df, function(x) is.factor(x) || is.ordered(x) || is.logical(x), logical(1)))
+    df[, to_chars] <- sapply(df[, to_chars], as.character, USE.NAMES = FALSE)
```

Comment:
df
```r
col1   col2
111  FALSE
222  TRUE
333  TRUE
444  TRUE
```
to_chars
```r
FALSE TRUE
```

when you lapply, you recieve
```r
[[1]]
[1] "FALSE"
[[2]]
[1] "TRUE"
[[3]]
[1] "TRUE"
[[4]]
[1] "TRUE"
```
and when you do `df[..., ] <- lapply(...)`
you end up with
```r
col1   col2
111  FALSE
222  FALSE
333  FALSE
444  FALSE
```

Funny, huh